### PR TITLE
feat: pending document ids with collection

### PR DESCRIPTION
**Pull request by jayahariv**
* Add implementation to include the pending doc ids implementation with collection
* Update unit tests to validate both methods

### CODE REVIEW

File: Objective-C/CBLReplicator.mm [hunk 1]
```diff
@@ -471,14 +479,26 @@ - (void) setProgressLevel: (CBLReplicatorProgressLevel)level {
 - (BOOL) isDocumentPending: (NSString *)documentID
                 collection: (CBLCollection *)collection
                      error: (NSError**)error {
+    CBLAssertNotNil(collection);
     
-    // TODO: Add implementation 
+    CBLStringBytes name(collection.name);
+    CBLStringBytes scopeName(collection.scope.name);
+    C4CollectionSpec c4spec = { .name = name, .scope = scopeName };
```
Comment (pasin) [inline Objective-C/CBLReplicator.mm:486]: Optionally, you could made an internal properly to get the collection spec from the collection object (e.g. collection.spec).

File: Objective-C/CBLReplicator.mm [hunk 2]
```diff
@@ -471,14 +479,26 @@ - (void) setProgressLevel: (CBLReplicatorProgressLevel)level {
...
 }
 
 - (BOOL) isDocumentPending: (NSString*)documentID error: (NSError**)error {
     CBLAssertNotNil(documentID);
+    
+    return [self isDocumentPending: documentID c4Spec: kC4DefaultCollectionSpec error: error];
+}
+
+// internal
+- (BOOL) isDocumentPending:(NSString *)documentID
```
Comment (pasin) [inline Objective-C/CBLReplicator.mm:498]: Why doesn't this get implemented in `- (BOOL) isDocumentPending: (NSString*)documentID collection: error: (NSError**)error` itself?

File: Objective-C/CBLReplicator.mm [hunk 3]
```diff
@@ -423,13 +424,21 @@ - (void) setProgressLevel: (CBLReplicatorProgressLevel)level {
...
-    return nil;
+    return [self pendingDocumentIDsForCollectionSpec: c4spec error: error];
 }
 
 - (NSSet<NSString*>*) pendingDocumentIDs: (NSError**)error {
+    return [self pendingDocumentIDsForCollectionSpec: kC4DefaultCollectionSpec error: error];
+}
+
+// internal
+- (NSSet<NSString*>*) pendingDocumentIDsForCollectionSpec: (C4CollectionSpec)spec error: (NSError**)error {
```
Comment (pasin) [inline Objective-C/CBLReplicator.mm:441]: Why doesn't this get implemented in `- (NSSet<NSString*>*) pendingDocumentIDsForCollection: (CBLCollection *)collection error: (NSError**)error` itself? Then the old pendingDocumentIDs will be called into the new one.

**Review by pasin — CHANGES_REQUESTED** [review]

File: Objective-C/CBLReplicator.mm (hunk 3 above)
Comment (jayahariv) [reply Objective-C/CBLReplicator.mm:441]: it was to avoid getting the default collection but use the default collection spec.

**Review by pasin — APPROVED** [review]

**Pull request merged**